File: experiments/020-PythonVsJava/cells/java-gradle_1-Empty_free/2019-2c-parcial2-masked/20260921-214358-38088/output/project/src/main/java/surveycrawler/AdvanceCommand.java
package surveycrawler;

public class AdvanceCommand extends RepeatableCrawlerCommand {

    // testing

    @Override
    public boolean isFor(char aCommand) {
        return aCommand == 'a';
    }

    // executing

    @Override
    protected void executeOnce(SurveyCrawler aCrawler) {
        aCrawler.advance();
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2019-2c-parcial2-masked/20260921-213927-34997/analysis/own-tests-project/src/surveycrawler/advance_command.py
from surveycrawler.crawler_command import CrawlerCommand


class AdvanceCommand(CrawlerCommand):

    # command character

    @classmethod
    def character(cls):
        return 'a'

    # executing

    def execute(self, a_crawler):
        a_crawler.advance()

    def undo(self, a_crawler):
        a_crawler.undo_advance()
